__tests__/sources.ts

```diff
@@ -421,6 +434,71 @@ describe('query sources', () => {
       'http://image.com/header',
     );
   });
+
+  const saveMembers = (sourceId: string, users: string[]) => {
+    const repo = con.getRepository(SourceMember);
+    const members = users.map((userId) =>
+      repo.create({
+        userId,
+        sourceId,
+        referralToken: randomUUID(),
+        role: SourceMemberRoles.Member,
+      }),
+    );
+
+    return repo.save(members);
+  };
+
+  it('should return public squads ordered by members count', async () => {
+    await prepareSquads();
+    await saveFixtures(con, Source, [sourcesFixture[2]]);
+    await con.getRepository(SourceMember).delete({ sourceId: Not('null') });
+    await con.getRepository(Source).update(
+      { id: Not('null') },
+      {
+        type: SourceType.Squad,
+        flags: updateFlagsStatement({ totalMembers: 0 }),
+      },
+    );
+    await saveMembers('a', ['1', '2']);
+    await saveMembers('b', ['1']);
+    await saveMembers('c', ['1', '2', '3']);
+
+    const query = QUERY({ first: 10, sortByMembersCount: true });
+    const res = await client.query(query);
+    expect(res.errors).toBeFalsy();
+
+    const [cSquad, aSquad, bSquad] = res.data.sources.edges;
+    expect(cSquad.node.id).toEqual('c');
+    expect(aSquad.node.id).toEqual('a');
+    expect(bSquad.node.id).toEqual('b');
+    const isBiggest = res.data.sources.edges
+      .filter(({ node }) => node.id !== 'c')
+      .every(
+        ({ node }) => cSquad.node.flags.totalMembers > node.flags.totalMembers,
+      );
```

Comment:
This is a bit clunky evaluation because you are only checking if first not is the biggest but interim notes can still be out of order, better to do something like `expect(res.data.sources.edges.map(item => item.id)).toBe(['a', 'b'])`

Reply:
Same for below test.

Reply:
I created a function to determine whether the list is sorted or not. Would love to hear what you think of it.

Reply:
I think it over complicates, in tests you don't want abstractions, you want a point blank comparison, so if you are expecting 4 elements in order we should just check that. 

```
expect(res.data.sources.edges.map(({ node }) => node.id)).toEqual([
  'c',
  'a',
  'b',
  'squad',
])
```

Reply:
Cool. Let's do that.

Reply:
My worry is that it might produce a flaky test, hence introduced a strict function.

Reply:
Yep, I am getting it now locally on the unordered list.

Reply:
> My worry is that it might produce a flaky test, hence introduced a strict function.

Its not really strict but its a function that evaluates what it gets vs what test expects. If its flaky then we need to fix the fixtures. For example you can introduce uniquely IDed members as here https://github.com/dailydotdev/daily-api/pull/2236#discussion_r1750173450

Reply:
I mean it is strict that the list must not be ordered which is one of the ways we can check whether they received the props or not that we are trying to identify. Although, tests should be deterministic, and the root cause it fails is yes, the fixtures.